lib/src/signed_video_tlv.c

```diff
@@ -902,6 +904,56 @@ tlv_find_tag(signed_video_t *self,
   return NULL;
 }
 
+svi_rc
+tlv_find_tag_and_decode_recurrent_tags(signed_video_t *self,
+    const uint8_t *tlv_data,
+    size_t tlv_data_size,
+    bool with_ep)
+{
+  const uint8_t *tlv_data_ptr = tlv_data;
+  const uint8_t *latest_tag_location = NULL;
+
+  if (!self || !tlv_data || tlv_data_size == 0) return SVI_INVALID_PARAMETER;
+
+  svi_rc status = SVI_UNKNOWN;
+  uint16_t last_two_bytes = LAST_TWO_BYTES_INIT_VALUE;
+  while (tlv_data_ptr < tlv_data + tlv_data_size) {
+    latest_tag_location = tlv_data_ptr;
+    // Read the tag
+    sv_tlv_tag_t this_tag = read_byte(&last_two_bytes, &tlv_data_ptr, with_ep);
+    // If it is a recurrent tag then decode it
+    if (!tlv_tuples[this_tag].is_always_present) {
+      size_t tlv_header_size = 0;
+      size_t length = 0;
+      sv_tlv_tag_t tag_check = 0;
+      status = decode_tlv_header(latest_tag_location, &tlv_header_size, &tag_check, &length);
```

Comment:
Can't we use decode_tlv_header already above when extracting the tag?

```
while (tlv_data_ptr < tlv_data + tlv_data_size) {
  size_t tlv_header_size = 0;
  size_t length = 0;
  sv_tlv_tag_t this_tag = UNDEFINED_TAG;
  status = decode_tlv_header(tlv_data_ptr, &tlv_header_size, &this_tag, &length);
  tlv_data_ptr += tlv_header_size;
  if (!tlv_tuples[this_tag].is_always_present) {
    sv_tlv_decoder_t decoder = get_decoder(this_tag);
    status = decoder(self, tlv_data_ptr, length);
  }
  tlv_data_ptr += length;
}
```
and handle return values.